core/src/ru/spbau/blackout/network/AndroidClientThread.java

```diff
@@ -0,0 +1,92 @@
+package ru.spbau.blackout.network;
+
+import com.badlogic.gdx.Gdx;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.Socket;
+import java.net.UnknownHostException;
+
+import ru.spbau.blackout.BlackoutGame;
+import ru.spbau.blackout.screens.MenuScreen;
+import ru.spbau.blackout.screens.MultiplayerTable;
+import ru.spbau.blackout.screens.PlayScreenTable;
+
+import static ru.spbau.blackout.BlackoutGame.HOST_NAME;
+import static ru.spbau.blackout.BlackoutGame.PORT_NUMBER;
+
+public class AndroidClientThread extends Thread {
```

Comment:
Not necessarily. It's just some process communicating with other processes (which is what a Runnable is). A request to stop coming from another process is just some form of communication. I'd use a flag (and ensure proper synchronization).